staging/vhost-device-sound/src/stream.rs

```diff
@@ -108,12 +104,11 @@ pub enum PCMState {
 
 macro_rules! set_new_state {
     ($new_state_fn:ident, $new_state:expr, $($valid_source_states:tt)*) => {
-        pub fn $new_state_fn(&mut self) -> Result<()> {
-            if !matches!(self, $($valid_source_states)*) {
-                return Err(Error::InvalidStateTransition(*self, $new_state));
+        pub fn $new_state_fn(&mut self) {
+            if !matches!(*self, $($valid_source_states)*) {
+                log::error!("{}", Error::InvalidStateTransition(*self, $new_state));
```

Comment:
As far as I understand, you have relaxed this requirement to log an error instead of returning an error when the outgoing transition is invalid. I would like to know what the use-case is for this. In which case is the guest driver transiting to an invalid state? Thanks.

Reply:
Hello Matias, see commit message in https://github.com/rust-vmm/vhost-device/pull/476/commits/4495818ef71539cd717a2a34c0c48db3d36bd058

Reply:
It's not invalid (from the ALSA pov) for a user space app to do invalid transitions it seems, I discovered this when using libsox. And since the transitions are non-normative in the spec, we shouldn't error on an unexpected transition

Reply:
Read it, thanks. I think it is better to log in instead of error. I'm wondering if it's a driver issue. When testing, what version of Linux did you use?

Reply:
Mmm I don't think so, there's no state handling in the driver.